src/client/Microsoft.Identity.Client/Instance/Discovery/RegionAndMtlsDiscoveryProvider.cs

```diff
@@ -73,7 +73,18 @@ private static string GetRegionalizedEnvironment(Uri authority, string region, R
             }
 
             requestContext.Logger.Info(() => $"[Region discovery] Regionalized Environment is : {region}.{host}. ");
-            return $"{region}.{host}";
+
+            // Decide whether to use MTLS or standard environment
+            if (!requestContext.UseMtlsPop)
```

Comment:
In region is configured by app developer, then the regional env will be returned at line 65. So it will ignore mTLS endpoint. 

This shows there is insufficient unit testing in this area.

Reply:
added a test but don't think it is the correct one, looking into it